pydoctor/astbuilder.py

```diff
@@ -507,14 +509,13 @@ def _get_all_args() -> Iterator[ast.arg]:
             kwargs = base_args.kwarg
             if kwargs:
                 yield kwargs
-        def _get_all_ast_annotations() -> Iterator[Tuple[str, ast.expr]]:
+        def _get_all_ast_annotations() -> Iterator[Tuple[str, Optional[ast.expr]]]:
             for arg in _get_all_args():
-                if arg.annotation:
-                    yield arg.arg, arg.annotation
+                yield arg.arg, arg.annotation
             returns = func.returns
             if returns:
                 yield 'return', returns
-        return {name: self._unstring_annotation(value)
+        return {name: None if value is None else self._unstring_annotation(value)
```

Comment:
The reason for keeping the `None` values is that the presence of the key is also information: we'll need this later to warn about documented non-existing parameters. I'll see where I can add some comments in the code, because this indeed isn't clear from the context.